client/src/components/ColumnArea.js

```diff
@@ -15,40 +17,56 @@ import {
 import CreateColumnButton from "../components/CreateColumnButton";
 import CreateBoardColumn from "./TitleInputModal";
 
-//materia-ui
-
 const ColumnArea = props => {
   const classes = useStyles(props);
   const { value1, dashboardTitles } = useContext(UserContext);
   let [taskState, setTaskState] = value1;
-  let [dbTitles] = dashboardTitles;
-
+  let [dbTitles, setdbTitles] = dashboardTitles;
   const [open, setOpen] = useState(false);
+  let dashboardId = (taskState && taskState._id) || props.match.params.dashboardId;
 
-  let dashboardId =
-    (taskState && taskState._id) || props.match.params.dashboardId;
+  console.log(props);
 
   useEffect(() => {
-    if (
-      Object.entries(dbTitles).length === 0 &&
-      dashboardId === "createboard"
-    ) {
+    if (Object.entries(dbTitles).length === 0 && dashboardId === "createboard") {
       setOpen(true);
       return;
     }
-  }, []);
+    if (getCurrentBoard()) {
+      getDashboard(getCurrentBoard(), res => {
+        if (!res) {
+          localStorage.removeItem("dashboard");
+          props.history.push(`/dashboards/createboard`);
+          return;
+        }
+
```

Comment:
this logic is already in top nav component. now its setting all the data twice which isnt good for memory

Reply:
Thanks. I removed!